irispy/new_sji.py

```diff
@@ -0,0 +1,179 @@
+import astropy.units as u
+from astropy.wcs import WCS
+from sunpy.time import parse_time
+from sunpy import config
+from irispy import iris_tools
+
+__all__ = ['SJICube']
+
+TIME_FORMAT = config.get("general", "time_format")
+
+# the following value is only appropriate for byte scaled images
+BAD_PIXEL_VALUE = -200
+# the following value is only appropriate for unscaled images
+BAD_PIXEL_VALUE_UNSCALED = -32768
+
+
+class SJICube(NDCube):
+    """
+    SJICube
+
+    Class representing SJI Images described by a single WCS
+
+    Parameters
+    ----------
+    data: `numpy.ndarray`
+        The array holding the actual data in this object.
+
+    wcs: `ndcube.wcs.wcs.WCS`
+        The WCS object containing the axes' information
+
+    unit : `astropy.unit.Unit` or `str`
+        Unit for the dataset.
+        Strings that can be converted to a Unit are allowed.
+
+    meta : dict-like object
+        Additional meta information about the dataset.
+
+    uncertainty : any type, optional
+        Uncertainty in the dataset. Should have an attribute uncertainty_type
+        that defines what kind of uncertainty is stored, for example "std"
+        for standard deviation or "var" for variance. A metaclass defining
+        such an interface is NDUncertainty - but isn’t mandatory. If the
+        uncertainty has no such attribute the uncertainty is stored as
+        UnknownUncertainty.
+        Defaults to None.
+
+    mask : any type, optional
+        Mask for the dataset. Masks should follow the numpy convention
+        that valid data points are marked by False and invalid ones with True.
+        Defaults to None.
+
+    extra_coords : iterable of `tuple`s, each with three entries
+        (`str`, `int`, `astropy.units.quantity` or array-like)
+        Gives the name, axis of data, and values of coordinates of a data axis
+        not included in the WCS object.
+
+    copy : `bool`, optional
+        Indicates whether to save the arguments as copy. True copies every
+        attribute before saving it while False tries to save every parameter
+        as reference. Note however that it is not always possible to save the
+        input as reference.
+        Default is False.
+
+    Examples
+    --------
+    >>> from irispy import sji
+    >>> from irispy.data import sample
+    >>> sji = read_iris_sji_level2_fits(sample.SJI_CUBE_1400)
+    """
+
+    def __repr__(self):
+        return (
+            """
+    SJICube
+    ---------
+    Observatory:\t\t {obs}
+    Instrument:\t\t\t {instrument}
+    Bandpass:\t\t\t {bandpass}
+    Obs. Start:\t\t\t {date_start:{tmf}}
+    Obs. End:\t\t\t {date_end:{tmf}}
+    Instance Start:\t\t {instance_start:{tmf}}
+    Instance End:\t\t {instance_end:{tmf}}
+    Total Frames in Obs.:\t {frame_num}
+    IRIS Obs. id:\t\t {obs_id}
+    IRIS Obs. Description:\t {obs_desc}
+    Cube dimensions:\t\t {dimensions}
+    Axis Types:\t\t\t {axis_types}
+    """.format(obs=self.meta["TELESCOP"],
```

Comment:
Adding new reminder to change ```self.meta["X"]``` to ```self.meta.get("X", None)``` as the old comment was marked as outdated.

Reply:
I did not change it because I was changing it on reading function, but now, with None, I have to modify it.

Reply:
If the header does not contain a value, the reading function will translate it to None. So we don't need to add this on repr.

Reply:
I found an error on the dates reading. If date is None, repr can not display a value due to the time format. Do you have an idea to solve it ? 

I think I need to modify the time format in reading function but I don't know how...

Reply:
That's incorrect.  ```self.meta.get("X", None)``` is still needed here.  Otherwise you're assuming that the only way an ```SJICube``` will ever be created is via ```read_iris_sji_level2_fits```.  This is an unnecessarily restrictive assumption that means ```SJICube``` is not fully independent of ```read_iris_sji_level2_fits```, which of course was the whole point of writing a separate reading function in the first place.

Reply:
Could you provide more detail on the dates error?  I'm not sure what you mean.

Reply:
Sure, I forget the fact we can use other files ... sorry 

If I put `None` in dates (if there is no dates on fits file) as can do the reading function, my meta for time is `None` and `None` can't be in a time format with `tmf=TIME_FORMAT` and `TIME_FORMAT = config.get("general", "time_format")`. This is just a try but this is not working for `__repr__`. This is hard to explain ...

Reply:
I forget to change for `.get("X", None)` before the PR, but it's modified now and will be available for next PR : I don't want to spam PR for each modification.

Reply:
```.get("X", None)``` should be changed in this PR as it was originally to define the ```SJICube``` with a ```__repr__```.  Don't worry about spamming the PR.  That's how the system is set up :)

Reply:
`.get("X", None)` added on new PR